lucene/test-framework/src/java/org/apache/lucene/util/LuceneTestCase.java

```diff
@@ -613,6 +613,7 @@ public static TestRuleIgnoreAfterMaxFailures replaceMaxFailureRule(TestRuleIgnor
     RuleChain r = RuleChain.outerRule(new TestRuleIgnoreTestSuites())
       .around(ignoreAfterMaxFailures)
       .around(suiteFailureMarker = new TestRuleMarkFailure())
+      .around(new VerifyTestClassNamingConvention())
```

Comment:
question: would this convention automatically and always apply to all classes derived from `LuceneTestCase` including any non-`org.apache` name spaces or would it be possible to opt-out (without an exclusion list) somehow for custom code that might perhaps have chosen a different convention?

Reply:
This is just code, anything can be changed... In the example I wrote it can't be turned off.